damnit/gui/editor.py

```diff
@@ -51,7 +57,16 @@ def run(self):
                 ctx_path = Path(ctx_file.name)
                 ctx_path.write_text(self.code)
 
-                ctx, error_info = get_context_file(ctx_path, self.context_python)
+                try:
+                    ctx, error_info = self.context_getter(ctx_path, self.context_python)
+                except:
+                    # Not a failure to evalute the context file, but a failure
+                    # to *attempt* to evaluate the context file (e.g. because of
+                    # a missing dependency).
+                    help_msg = "# This is a partial error, please check the terminal for the full error message and ask for help from DA or the DOC."
+                    traceback_str = f"{help_msg}\n\n{traceback.format_exc()}"
+                    self.check_result.emit(ContextTestResult.ERROR, traceback_str, 0, 0, "")
```

Comment:
```suggestion
                    self.check_result.emit(ContextTestResult.ERROR, traceback_str, 0, 0, self.code)
```

The string of code checked is used as part of saving, so the downstream logic knows that the check has failed on the code we were planning to save. Doesn't do much at the moment, but it's easy to preserve.

Reply:
Fixed in 2cd5163.